### PR TITLE
Complete NAPI compatibility

### PR DESCRIPTION
**Pull request by JunielKatarn**
*Work in progress*

Allows creating an ABI-safe NAPI-based V8 JSI environment.

- Add MakeNapiJsiRuntime calling conventions
- Pop-Location on error
- Have napi_ext_create_env use napi_ext_env_settings* as firs argument
- Add flags to napi_ext_env_settings

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/v8-jsi/pull/64)

### CODE REVIEW

File: src/public/NapiJsiRuntime.h [hunk 1]
```diff
@@ -11,10 +11,10 @@
 #include <memory>
 #include "js_native_ext_api.h"
 
-namespace napijsi {
+namespace Microsoft { namespace JSI {
 
-std::unique_ptr<facebook::jsi::Runtime> MakeNapiJsiRuntime(napi_env env) noexcept;
+std::unique_ptr<facebook::jsi::Runtime> __cdecl MakeNapiJsiRuntime(napi_env env) noexcept;
```
Comment (tudorms) [inline src/public/NapiJsiRuntime.h:16]: This may need a `__declspec(dllexport)`

File: src/public/js_native_ext_api.h [hunk 2]
```diff
@@ -28,8 +28,71 @@ typedef struct napi_ext_ref__ *napi_ext_ref;
...
 
+// A callback to run task
+typedef void (*napi_ext_task_callback)(napi_env env, void *task_data);
+
+// A callback to schedule a task
+typedef void (*napi_ext_schedule_task_callback)(
+    napi_env env,
+    napi_ext_task_callback task_cb,
+    void *task_data,
+    uint32_t delay_in_msec,
```
Comment (tudorms) [inline src/public/js_native_ext_api.h:39]: >uint32_t delay_in_msec, [](http://example.com/codeflow?start=4&length=23)

(nit) delay no longer needed, we could simplify this and NapiJSITaskRunner a little bit (but we could do it later as well).

**Review by tudorms — APPROVED** [review]
:shipit:

**Comment by JunielKatarn** [comment]
Merged accidentally, apparently the auto-merge stuck even after removing the label.

That said, the change is correct; only incomplete.

**Pull request merged**